shopinvader_api_sale/routers/sales.py

```diff
@@ -39,30 +45,40 @@ def search(
 def get(
     sale_id: int,
     env: Annotated[api.Environment, Depends(authenticated_partner_env)],
+    partner: Annotated[ResPartner, Depends(authenticated_partner)],
 ) -> Sale:
     """
     Get sale order of authenticated user with specific sale_id
     """
     return Sale.from_sale_order(
-        env["shopinvader_api_sale.sales_router.helper"]._get(sale_id)
+        env["shopinvader_api_sale.sales_router.helper"]._get(partner, sale_id)
     )
 
 
 class ShopinvaderApiSaleSalesRouterHelper(models.AbstractModel):
     _name = "shopinvader_api_sale.sales_router.helper"
     _description = "Shopinvader Api Sale Service Helper"
 
-    @property
-    def filtered_model(self):
+    def _get_domain_adapter(self, partner):
```

Comment:
@sebastienbeau To ease extension, partner_id should be a property of ShopinvaderApiSaleSalesRouterHelper

```

class ShopinvaderApiSaleSalesRouterHelper(models.AbstractModel):

    partner = fields.One2many("res.partner")



# To use-it:

env["shopinvader_api_sale.sales_router.helper"]._new({"partner": partner}).get(sale_id)

```

Reply:
I was hesitating with this solution ;) I will change it